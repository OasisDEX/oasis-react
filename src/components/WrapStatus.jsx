import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import styles from './WrapStatus.scss';
import {
  TOKEN_WRAP_STATUS_AWAITING_TRANSFER_TO_BROKER_APPROVAL,
  TOKEN_WRAP_STATUS_AWAITING_TRANSFER_TO_WRAPPER_CONTRACT_APROVAL,
  TOKEN_WRAP_STATUS_TRANSFER_TO_BROKER_COMPLETE,
  TOKEN_WRAP_STATUS_TRANSFER_TO_BROKER_PENDING, TOKEN_WRAP_STATUS_WRAP_COMPLETE,
} from '../store/reducers/wrapUnwrap';


const propTypes = PropTypes && {
  activeTokenWrapStatus: PropTypes.string
};
const defaultProps = {};


class WrapStatus extends PureComponent {

  getStatus() {
    const { activeTokenWrapStatus } = this.props;
    switch (activeTokenWrapStatus) {
      case TOKEN_WRAP_STATUS_AWAITING_TRANSFER_TO_BROKER_APPROVAL: break;
      case TOKEN_WRAP_STATUS_TRANSFER_TO_BROKER_PENDING: break;
      case TOKEN_WRAP_STATUS_TRANSFER_TO_BROKER_COMPLETE: break;
      case TOKEN_WRAP_STATUS_AWAITING_TRANSFER_TO_WRAPPER_CONTRACT_APROVAL: break;
      case TOKEN_WRAP_STATUS_WRAP_COMPLETE: break;
      default: return null;
    }
  }

  render() {
    return (
      <div className={styles.base}>
        {this.getStatus()}
      </div>
    );
  }
}

WrapStatus.displayName = 'WrapStatus';
WrapStatus.propTypes = propTypes;
WrapStatus.defaultProps = defaultProps;
export default WrapStatus;
