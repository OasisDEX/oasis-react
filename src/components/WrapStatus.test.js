/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import WrapStatus from './WrapStatus';


describe('(Component) WrapStatus', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <WrapStatus {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
