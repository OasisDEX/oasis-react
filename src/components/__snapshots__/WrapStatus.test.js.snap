// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) WrapStatus should render 1`] = `<div />`;
